src/backend/ctrl_flow_struct/graph_utils/ncd.rs

```diff
@@ -0,0 +1,181 @@
+//! https://doi.org/10.1016/0196-6774(92)90063-I
```

Comment:
Explain what this module is as doc comments, context as to why this is important and a requirement for CF restructuring. May be outline the algorithm, like the very high-level details / intuition (cause it is pay-walled and not everyone can access that link).